src/persist/src/operators/mod.rs

```diff
@@ -24,3 +39,125 @@ pub(crate) fn split_ok_err<K, V>(
         (Err(err), ts, diff) => Err((err, ts, diff)),
     }
 }
+
+/// A helper type to integrate yielding to timely with async futures. Its intended to be used with
+/// a top level while loop. Timely will automatically make the returned futures resolve when there
+/// are progress or data updates to be processed by the operator
+///
+/// ```ignore
+/// async fn example(s: Scheduler) {
+///     while scheduler.yield_now().await {
+///     }
+/// }
+/// ```
+#[derive(Debug)]
+pub struct Scheduler(Rc<Cell<bool>>);
```

Comment:
nit: it might be a touch easier to read all this if this field was named (scheduled? yielded?)